### PR TITLE
Remove OSx+Mono skips related to aspnet/External#50

### PR DESCRIPTION
**Pull request by dougbu**
- CoreFx worked around the underlying issue
- add `PlatformNormalizer.NormalizeContent()` call in previously-disabled test

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.Core.Test/RequireHttpsAttributeTests.cs [hunk 1]
```diff
@@ -45,12 +44,8 @@ public static IEnumerable<object[]> RedirectToHttpEndpointTestData
                 data.Add("localhost", "/pathbase", "/path", "?foo=bar", "https://localhost/pathbase/path?foo=bar");
 
                 // Encode some special characters on the url.
-                // Two paths hit aspnet/External#50 with Mono on Mac.
-                if (!TestPlatformHelper.IsMac || !TestPlatformHelper.IsMono)
-                {
-                    data.Add("localhost", "/path?base", null, null, "https://localhost/path%3Fbase");
-                    data.Add("localhost", null, "/pa?th", null, "https://localhost/pa%3Fth");
-                }
+                data.Add("localhost", "/path?base", null, null, "https://localhost/path%3Fbase");
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/RequireHttpsAttributeTests.cs]: No longer need to do `data.Add` anymore. Can change this entire set to be like the other data properties we have:

`return new TheoryData....`

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by dougbu** [comment]
8189d85

**Pull request closed without merging**